src/lib.rs

```diff
@@ -1959,11 +1958,13 @@ fn truer_exec(filename: PathBuf, pretend_to_be_vanilla: bool) -> Result<(), Stri
             }
             if let Ok(event) = SOKU_LOOP_EVENT.lock() {
                 // if the last signal hasn't been reset by the loop
-                if let Some(event) = *event
-                    && WaitForSingleObject(HANDLE(event), 0).0 == 0
-                {
-                    println!("frame costed too much time!");
-                    WARNING_FRAME_LOST_COUNTDOWN.store(115, Relaxed);
+                if let Some(event) = *event {
+                    if WaitForSingleObject(HANDLE(event), 0).0 == 0 {
+                        println!("frame costed too much time!");
+                        WARNING_FRAME_LOST_COUNTDOWN.store(115, Relaxed);
+                    } else if WARNING_FRAME_LOST_COUNTDOWN.load(Relaxed) != 0 {
+                        WARNING_FRAME_LOST_COUNTDOWN.fetch_sub(1, Relaxed);
+                    }
                 } else if WARNING_FRAME_LOST_COUNTDOWN.load(Relaxed) != 0 {
                     WARNING_FRAME_LOST_COUNTDOWN.fetch_sub(1, Relaxed);
                 }
```

Comment:
Here `WARNING_FRAME_LOST_COUNTDOWN.fetch_sub(1, Relaxed);` is written twice. Maybe keep the old version instead?